chalicelib/checks/wfr_checks.py

```diff
@@ -1,1448 +1,1456 @@
+        get_and_add_uuids(found, meta_workflow_run_uuids)
+        get_and_add_field(found, "title", meta_workflow_run_titles)
+        meta_workflow_runs_not_found += not_found
+    else:
+        query = "/search/?type=MetaWorkflowRun&field=uuid&field=title" + "".join(
+            ["&final_status=" + st for st in FINAL_STATUS_TO_RESET]
+        )
+        search_response = ff_utils.search_metadata(query, key=connection.ff_keys)
+        get_and_add_uuids(search_response, meta_workflow_run_uuids)
+        get_and_add_field(search_response, "title", meta_workflow_run_titles)
+    msg = "%s MetaWorkflowRun(s) have failed WorkflowRuns to reset" % len(
+        meta_workflow_run_uuids
+    )
+    check.summary = msg
     check.brief_output.append(msg)
-    check.full_output['metawfrs_that_failed'] = {'titles': metawfr_titles, 'uuids': metawfr_uuids}
-    if kwargs.get('reset_all_failed'):
-        check.full_output['options'] = ['reset_all_failed']
-
+    check.full_output["meta_workflow_runs"] = {
+        "uuids": meta_workflow_run_uuids,
+        "titles": meta_workflow_run_titles,
+    }
+    if meta_workflow_runs_not_found:
+        msg = "%s MetaWorkflowRun identifiers could not be found" % len(
+            meta_workflow_runs_not_found
+        )
+        check.brief_output.append(msg)
+        check.full_output["not_found"] = meta_workflow_runs_not_found
+        check.status = "WARN"
+    if not meta_workflow_run_uuids:
+        check.allow_action = False
     return check
 
 
 @action_function()
 def reset_failed_metawfrs(connection, **kwargs):
+    """Reset all failed WorkflowRuns on MetaWorkflowRuns."""
     start = datetime.utcnow()
-    action = ActionResult(connection, 'reset_failed_metawfrs')
-    action_logs = {'runs_reset': []}
-    my_auth = connection.ff_keys
-    env = connection.ff_env
-    check_result = action.get_associated_check_result(kwargs).get('full_output', {})
-    action_logs['check_output'] = check_result
-    metawfr_uuids = check_result.get('metawfrs_that_failed', {}).get('uuids', [])
-    reset_all_failed = 'reset_all_failed' in check_result.get('options', [])  # pass ths option from check
-
-    random.shuffle(metawfr_uuids)  # if always the same order, we may never get to the later ones.
-    for metawfr_uuid in metawfr_uuids:
+    action, check_result = initialize_action(connection, kwargs)
+    action.description = "Reset all failed WorkflowRuns on MetaWorkflowRuns"
+
+    success = []
+    error = {}
+    meta_workflow_runs = check_result.get("meta_workflow_runs", {})
+    meta_workflow_run_uuids = meta_workflow_runs.get("uuids", [])
+    random.shuffle(meta_workflow_run_uuids)  # Ensure later ones hit within time limits
+    for meta_workflow_run_uuid in meta_workflow_run_uuids:
         now = datetime.utcnow()
-        if (now-start).seconds > lambda_limit:
-            action.description = 'Did not complete action due to time limitations'
+        if (now - start).seconds > lambda_limit:
```

Comment:
this could probably be extracted to a helper function since its repeated so often, but up to you.